puma/state_graph/action.py

```diff
@@ -1,6 +1,53 @@
 import inspect
+from typing import Callable
 
+from build.lib.puma.state_graph.puma_driver import PumaClickException
 from puma.state_graph.state import State
+from puma.state_graph.utils import safe_func_call, filter_arguments
+
+
+def _assert_verify_with_function_is_valid(verify_with):
+    """
+    Validate that the 'verify_with' passed to the action is valid, else throw an error.
+
+    :param verify_with: the object passed as 'verify_with'
+    :raises TypeError: if the verification function is not a callable
+    """
+    if not isinstance(verify_with, Callable):
+        raise TypeError(f"'verify_with' must be a callable, instead is: {type(verify_with)}")
+
+    # TODO: enforce signature, e.g. must be func(driver, gtl_logger) and optional extra args?
```

Comment:
Will this `TODO` be addressed in this PR, or shall we create an issue for it?

Reply:
@btimbermont @Angelina-C 

Should the verification function receive a driver or the application itself? Or something different?